src/course-unit/course-xblock/CourseXBlock.jsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ActionRow, Card, Dropdown, Icon, IconButton, useToggle,
+} from '@openedx/paragon';
+import { EditOutline as EditIcon, MoreVert as MoveVertIcon } from '@openedx/paragon/icons';
+import { useIntl } from '@edx/frontend-platform/i18n';
+
+import DeleteModal from '../../generic/delete-modal/DeleteModal';
+import { scrollToElement } from '../../course-outline/utils';
+import messages from './messages';
+
+const CourseXBlock = ({
+  id, title, unitXBlockActions, shouldScroll, ...props
+}) => {
+  const courseXBlockElementRef = useRef(null);
+  const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useToggle(false);
+  const intl = useIntl();
+
+  const onXBlockDelete = () => {
+    unitXBlockActions.handleDelete(id);
+    closeDeleteModal();
+  };
+
+  useEffect(() => {
+    // if this item has been newly added, scroll to it.
+    if (courseXBlockElementRef.current && shouldScroll) {
+      scrollToElement(courseXBlockElementRef.current);
+    }
+  }, []);
+
+  return (
+    <div ref={courseXBlockElementRef} {...props}>
+      <Card className="mb-1">
+        <Card.Header
+          title={title}
+          actions={(
+            <ActionRow>
+              <IconButton
+                alt={intl.formatMessage(messages.blockAltButtonEdit)}
+                iconAs={EditIcon}
+                size="md"
+                onClick={() => {}}
+              />
+              <Dropdown>
```

Comment:
The functionality of the dropdown is great, but the text color should not be colored like it is a link. Additionally the hover state for the dropdown item should not be blue. Currently it creates a contrast error which is an accessibility concern.